django_mongodb/compiler.py

```diff
@@ -154,23 +159,42 @@ def _prepare_annotations_for_aggregation_pipeline(self, order_by):
         group.update(having_group)
         return group, replacements
 
-    def _get_group_id_expressions(self, order_by):
-        """Generate group ID expressions for the aggregation pipeline."""
-        group_expressions = set()
-        replacements = {}
+    def _get_group_expressions(self, order_by):
+        # The query.group_by is either None (no GROUP BY at all), True
+        # (group by select fields), or a list of expressions to be added
+        # to the group by.
+        if self.query.group_by is None:
+            return []
+        seen = set()
+        expressions = set()
+        if self.query.group_by is not True:
+            for expr in self.query.group_by:
+                if not hasattr(expr, "as_sql"):
+                    expr = self.query.resolve_ref(expr)
+                if isinstance(expr, Ref):
+                    if expr.refs not in seen:
+                        seen.add(expr.refs)
+                        expressions.add(expr.source)
+                else:
+                    expressions.add(expr)
+        for expr, _, alias in self.select:
+            # Skip members that are already grouped.
+            if alias not in seen:
+                expressions |= set(expr.get_group_by_cols())
         if not self._meta_ordering:
             for expr, (_, _, is_ref) in order_by:
+                # Skip references.
                 if not is_ref:
-                    group_expressions |= set(expr.get_group_by_cols())
-        for expr, *_ in self.select:
-            group_expressions |= set(expr.get_group_by_cols())
+                    expressions |= set(expr.get_group_by_cols())
         having_group_by = self.having.get_group_by_cols() if self.having else ()
         for expr in having_group_by:
-            group_expressions.add(expr)
-        if isinstance(self.query.group_by, tuple | list):
-            group_expressions |= set(self.query.group_by)
-        elif self.query.group_by is None:
-            group_expressions = set()
+            expressions.add(expr)
+        return self.collapse_group_by(expressions, having_group_by)
```

Comment:
The logic is taken from [here](https://github.com/mongodb-forks/django/blob/85c154da2f07485a1cdc4d886eee4c1a1ef56137/django/db/models/sql/compiler.py#L94) and the idea is to avoid repeating expressions and merge both aggregators.

Reply:
Re-reading I figured out that we haven't defined `allows_group_by_selected_pks` so there is not need to call this function (actually this function does nothing)